pkg/controller/network/resourcerecordset/controller_test.go

```diff
@@ -0,0 +1,365 @@
+	MockListResourceRecordSetsRequest   func(*route53.ListResourceRecordSetsInput) route53.ListResourceRecordSetsRequest
+}
+
+func (m *MockResourceRecordSetClient) ChangeResourceRecordSetsRequest(input *route53.ChangeResourceRecordSetsInput) route53.ChangeResourceRecordSetsRequest {
+	return m.MockChangeResourceRecordSetsRequest(input)
+}
+
+func (m *MockResourceRecordSetClient) ListResourceRecordSetsRequest(input *route53.ListResourceRecordSetsInput) route53.ListResourceRecordSetsRequest {
+	return m.MockListResourceRecordSetsRequest(input)
+}
+
+var (
+	mockExternalClient external
+	mockClient         MockResourceRecordSetClient
+
+	unexpectedItem resource.Managed
+)
+
+func TestMain(m *testing.M) {
+	mockClient = MockResourceRecordSetClient{}
+	mockExternalClient = external{
+		client: &mockClient,
+		kube: &test.MockClient{
+			MockUpdate: test.NewMockUpdateFn(nil),
+		},
+	}
+
+	os.Exit(m.Run())
+}
+
+func Test_Observe(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+
+	name := "x.x.x."
+	mockManaged := v1alpha3.ResourceRecordSet{}
+	meta.SetExternalName(&mockManaged, name)
+
+	incorrectName := "p.p.p."
+	mockManagedRRNotPresent := v1alpha3.ResourceRecordSet{
+		Spec: v1alpha3.ResourceRecordSetSpec{
+			ForProvider: v1alpha3.ResourceRecordSetParameters{
+				Name: &incorrectName,
+			},
+		},
+	}
+	meta.SetExternalName(&mockManagedRRNotPresent, name)
+
+	mockManagedRRPresent := v1alpha3.ResourceRecordSet{
+		Spec: v1alpha3.ResourceRecordSetSpec{
+			ForProvider: v1alpha3.ResourceRecordSetParameters{
+				Name: &name,
+			},
+		},
+	}
+	meta.SetExternalName(&mockManagedRRPresent, name)
+
+	mockExternal := &route53.ResourceRecordSet{
+		Name: aws.String(name),
+	}
+
+	var mockClientErr error
+	var itemsList []route53.ResourceRecordSet
+
+	mockClient.MockChangeResourceRecordSetsRequest = func(input *route53.ChangeResourceRecordSetsInput) route53.ChangeResourceRecordSetsRequest {
+		return route53.ChangeResourceRecordSetsRequest{
+			Request: &aws.Request{
+				HTTPRequest: &http.Request{},
+				Data:        &route53.ChangeResourceRecordSetsOutput{},
+				Error:       mockClientErr,
+			},
+		}
+	}
+
+	mockClient.MockListResourceRecordSetsRequest = func(input *route53.ListResourceRecordSetsInput) route53.ListResourceRecordSetsRequest {
+		return route53.ListResourceRecordSetsRequest{
+			Request: &aws.Request{
+				HTTPRequest: &http.Request{},
+				Data: &route53.ListResourceRecordSetsOutput{
+					ResourceRecordSets: itemsList,
+				},
+				Error: mockClientErr,
+			},
+		}
+	}
+
+	for _, tc := range []struct {
+		description               string
+		managedObj                resource.Managed
+		itemsReturned             []route53.ResourceRecordSet
+		clientErr                 error
+		expectedErrNil            bool
+		expectedResourceExist     bool
+		expectedResourceAvailable bool
+	}{
+		{
+			"upexpected managed resource should return error",
+			unexpectedItem,
+			nil,
+			nil,
+			false,
+			false,
+			false,
+		},
```

Comment:
I think these tests would be much easier to follow if we included field names in the structs.